### PR TITLE
Nextjs-Vite: Move vite-plugin-storybook-nextjs into the monorepo

### PR DESCRIPTION
**Pull request by ndelangen**
Closes #

## What I did

Spike: move [`vite-plugin-storybook-nextjs`](https://github.com/storybookjs/vite-plugin-storybook-nextjs) into the Storybook monorepo under `code/lib/vite-plugin-storybook-nextjs`, preserving full git history from the standalone repo.

Context from team discussion (Valentin): aside from historical experimental/release-cadence reasons, there is no strong reason to keep the package outside the monorepo anymore, and the old "tsup build flexibility" rationale is not a blocker.

### Import / history

- Rewrote the standalone repo history with `git filter-repo --to-subdirectory-filter code/lib/vite-plugin-storybook-nextjs`
- Merged with `--allow-unrelated-histories` so `git log -- code/lib/vite-plugin-storybook-nextjs/...` retains prior commits

### Monorepo adaptation

- Added NX/`build-config.ts` wiring (esbuild ESM build; browser + node mock export paths now both point at `.js`)
- Synced package version to Storybook (`10.6.0-alpha.5`) and registered it in `scripts/build/entry-configs.ts` + `versions.ts`
- Pointed `@storybook/nextjs-vite` at `workspace:*` and switched it to a normal ESM import (removed the `require()` workaround for the old dual CJS/ESM publish)
- Removed standalone-repo tooling (`pnpm`, changesets, biome, example app, GitHub workflows, tsup)
- Kept the npm package name `vite-plugin-storybook-nextjs` for continuity

### Follow-ups (intentionally out of this spike)

- Decide whether to eventually fold the plugin into `@storybook/nextjs-vite` itself (Jeppe/Michael comments) vs keep it as a separately published workspace package
- Archive/redirect the standalone repo after the first monorepo release that publishes this package
- Port open PRs from the standalone repo that are still valuable
- Broader cleanup of leftover `biome-ignore` comments / style nits

#### The changes in this PR are covered in the following automated tests:

- [ ] stories
- [x] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

> [!CAUTION]
> This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!

1. From repo root: `yarn nx compile vite-plugin-storybook-nextjs nextjs-vite` (or `yarn nx run-many -t compile -p vite-plugin-storybook-nextjs,nextjs-vite`)
2. Confirm unit tests pass: `yarn test code/lib/vite-plugin-storybook-nextjs`
3. Generate a Next.js Vite sandbox and start Storybook:
   ```bash
   yarn task sandbox --template nextjs-vite/default-ts --start-from auto
   ```
4. Open the sandbox Storybook and verify Next.js features still work (Image, Link, navigation mocks, fonts if present)
5. Optionally smoke portable stories / Vitest addon against that sandbox if you have it configured

Areas most likely to regress: mock alias resolution (`browser/mocks/*` vs `node/mocks/*`), ESM import of the plugin from `@storybook/nextjs-vite`, and production d.ts emit for the main entry.

### Documentation

- [x] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli-storybook/src/sandbox-templates.ts`
- [ ] Declare whether manual QA will be needed for this PR during the next release, through `qa:needed` or `qa:skip`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

  - `bug`: Internal changes that fixes incorrect behavior.
  - `maintenance`: User-facing maintenance tasks.
  - `dependencies`: Upgrading (sometimes downgrading) dependencies.
  - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
  - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
  - `documentation`: Documentation **only** changes. Will not show up in release changelog.
  - `feature request`: Introducing a new feature.
  - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
  - `other`: Changes that don't fit in the above categories.

   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/publish.yml) or locally with `gh workflow run --repo storybookjs/storybook publish.yml --field pr=<PR_NUMBER>`_

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Comment by valentinpalkovic** [comment]
@ndelangen, can you give me some pointers on what you had to change to make it work in the monorepo? I'll only review these parts then.

File: code/lib/vite-plugin-storybook-nextjs/src/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  default: loadJsConfigMock,
+}));
+
+vi.mock('next/dist/server/config.js', () => ({
+  default: loadConfigMock,
+}));
+
+vi.mock('vite-tsconfig-paths', () => ({
+  default: tsconfigPathsMock,
+}));
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/lib/vite-plugin-storybook-nextjs/src/index.test.ts:29]: Won't fix for this PR.

Imported tests already pass; rewriting every factory mock to `{ spy: true }` / `vi.mocked()` is a large style-only change with no behavior impact. Happy to do a follow-up cleanup pass for this package's Vitest style.

File: code/lib/vite-plugin-storybook-nextjs/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+  const nodeEnv = process.env.NODE_ENV;
+  const phase =
+    nodeEnv === 'development'
+      ? PHASE_DEVELOPMENT_SERVER
+      : nodeEnv === 'test'
+        ? PHASE_TEST
+        : PHASE_PRODUCTION_BUILD;
+
+  loadNextConfig(phase, resolvedDir).then(nextConfigResolver.resolve, nextConfigResolver.reject);
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/lib/vite-plugin-storybook-nextjs/src/index.ts:58]: Won't fix for this PR.

Pre-existing race around `loadNextConfig` rejection. Failures still surface when Vite awaits plugin/`config` consumers; adding a no-op handler here doesn't change the move's risk profile. Fine as a follow-up hardening item.

File: code/lib/vite-plugin-storybook-nextjs/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+      },
+      configResolved(config) {
+        if (isVitestEnv && !config.test?.browser?.enabled) {
+          // biome-ignore lint/style/noNonNullAssertion: test is available in the config
+          config.test!.setupFiles = [
+            require.resolve('./mocks/storybook.global.js'),
+            ...(config.test?.setupFiles ?? []),
+          ];
+        }
+      },
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/lib/vite-plugin-storybook-nextjs/src/index.ts:203]: Won't fix — false positive for this path.

`isVitestEnv` is Vitest-driven and Vitest supplies `config.test`. The non-null assertion matches that contract (and the existing biome-ignore).

File: code/lib/vite-plugin-storybook-nextjs/src/plugins/next-env/plugin.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  try {
+    // Next.js >= 15.4.0
+    getDefineEnv = (await import('next/dist/build/define-env.js')).getDefineEnv;
+    isNext1540 = true;
+  } catch (error) {
+    // Next.js < 15.4.0
+    getDefineEnv =
+      // @ts-expect-error - TODO: Ignoring because types are for >= 15.4.0
+      (await import('next/dist/build/webpack/plugins/define-env-plugin.js')).getDefineEnv;
+  }
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/lib/vite-plugin-storybook-nextjs/src/plugins/next-env/plugin.ts:30]: Won't fix for this PR.

The broad catch is intentional probing between Next ≥15.4 `define-env` and the legacy import path. Tightening to only "missing entry" would be a separate robustness change, not required for the monorepo move.

File: code/lib/vite-plugin-storybook-nextjs/src/plugins/next-font/google/get-font-face-declarations.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      ? cssCache.get(url)
+      : await fetchCSSFromGoogleFonts(url, fontFamily, true).catch(() => null);
+    if (!hasCachedCSS) {
+      cssCache.set(url, fontFaceCSS as string);
+    } else {
+      cssCache.delete(url);
+    }
+    if (fontFaceCSS === null) {
+      throw new Error(`Failed to fetch \`${fontFamily}\` from Google Fonts with URL: \`${url}\``);
+    }
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/lib/vite-plugin-storybook-nextjs/src/plugins/next-font/google/get-font-face-declarations.ts:56]: Won't fix for this PR.

Delete-on-hit matches Next's own `@next/font` loader cache behavior. Refetch still works; this isn't a regression from the move.

File: code/lib/vite-plugin-storybook-nextjs/src/plugins/next-font/plugin.ts [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+
+      let fontFaceDeclaration: FontFaceDeclaration | undefined;
+
+      const pathSep = path.sep;
+
+      if (sourceWithoutQuery.endsWith(['next', 'font', 'google', 'target.css'].join(pathSep))) {
+        fontFaceDeclaration = await getGoogleFontFaceDeclarations(fontOptions);
+      }
+
+      if (sourceWithoutQuery.endsWith(['next', 'font', 'local', 'target.css'].join(pathSep))) {
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/lib/vite-plugin-storybook-nextjs/src/plugins/next-font/plugin.ts:76]: Won't fix — not a practical break.

Vite/`pathe` normalize ids to `/`, so the `endsWith` checks match real inputs. The include pattern accepting `\` doesn't mean we see Windows separators at this point.

File: code/lib/vite-plugin-storybook-nextjs/src/plugins/next-font/utils/set-font-declarations-in-head.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    style.setAttribute('id', 'font-face-${id}');
+    style.innerHTML = fontDeclarations;
+    document.head.appendChild(style);
+
+    const classNamesCSS = \`${classNamesCSS}\`;
+    const classNamesStyle = document.createElement('style');
+    classNamesStyle.setAttribute('id', 'classnames-${id}');
+    classNamesStyle.innerHTML = classNamesCSS;
+    document.head.appendChild(classNamesStyle);
+  }
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/lib/vite-plugin-storybook-nextjs/src/plugins/next-font/utils/set-font-declarations-in-head.ts]: Fixed in 09373301096.

Guard now checks `font-face-${id}`, matching the id we actually set on the injected `<style>`.

File: code/lib/vite-plugin-storybook-nextjs/src/plugins/next-image/plugin.ts [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+  let hasVitePluginSvgr = false;
+  const postfixRE = /[?#].*$/s;
+  const filter = createFilter(
+    [
+      '**/*.{png,jpg,jpeg,gif,webp,avif,ico,bmp,svg}',
+      '**/*.{png,jpg,jpeg,gif,webp,avif,ico,bmp,svg}?*',
+      '**/*.{png,jpg,jpeg,gif,webp,avif,ico,bmp,svg}#*',
+    ],
+    options.excludeFiles
+  );
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/lib/vite-plugin-storybook-nextjs/src/plugins/next-image/plugin.ts:67]: Fixed in 7afb28757be.

`createFilter` now uses `options.includeFiles` when provided (defaults unchanged). `excludeFiles` already worked.

File: code/lib/vite-plugin-storybook-nextjs/src/plugins/next-mocks/alias/headers/headers.ts [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+let headersAdapterMock: HeadersAdapterMock;
+
+export const headers = () => {
+  if (!headersAdapterMock) {
+    headersAdapterMock = new HeadersAdapterMock();
+  }
+  return headersAdapterMock;
+};
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/lib/vite-plugin-storybook-nextjs/src/plugins/next-mocks/alias/headers/headers.ts:35]: Won't fix for this PR.

Async/`use()`-compatible headers/cookies mocks are a real Next 16 parity gap, but they're a heavy mock redesign and out of scope for moving the package into the monorepo. Follow-up material.

File: code/lib/vite-plugin-storybook-nextjs/src/plugins/next-mocks/alias/link/index.tsx [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+  const rawHref =
+    typeof href === 'object'
+      ? `${href.pathname || ''}${
+          href.query
+            ? `?${new URLSearchParams(href.query as Record<string, string>).toString()}`
+            : ''
+        }${href.hash || ''}`
+      : href;
+
+  const hrefString = typeof rawHref === 'string' ? normalizePathTrailingSlash(rawHref) : rawHref;
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/lib/vite-plugin-storybook-nextjs/src/plugins/next-mocks/alias/link/index.tsx:80]: Won't fix for this PR.

Valid mock fidelity gap (`UrlObject.search` dropped). Pre-existing; not introduced by the move. Tracking as follow-up mock parity, not blocking this PR.

File: code/lib/vite-plugin-storybook-nextjs/src/plugins/next-mocks/alias/navigation/index.ts [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+// mock utilities/overrides (as of Next v14.2.0)
+export const redirect = fn((url: string, type: RedirectType = 'push' as RedirectType): never => {
+  throw getRedirectError(url, type, RedirectStatusCode.SeeOther);
+}).mockName('next/navigation::redirect');
+
+export const permanentRedirect = fn(
+  (url: string, type: RedirectType = 'push' as RedirectType): never => {
+    throw getRedirectError(url, type, RedirectStatusCode.SeeOther);
+  }
+).mockName('next/navigation::permanentRedirect');
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/lib/vite-plugin-storybook-nextjs/src/plugins/next-mocks/alias/navigation/index.ts:90]: Won't fix for this PR.

Agreed the status code should be `PermanentRedirect` (308) for parity. One-line mock accuracy fix, but pre-existing and not needed for the move — leave for a mock-parity follow-up.

File: code/lib/vite-plugin-storybook-nextjs/src/plugins/next-mocks/alias/safe-stable-stringify/index.ts [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+ * Original file can be found here:
+ * https://github.com/vercel/next.js/blob/canary/packages/next/src/compiled/safe-stable-stringify/index.js
+ */
+export default function safeStableStringify(value: unknown): string {
+  try {
+    return JSON.stringify(value);
+  } catch {
+    return String(value);
+  }
+}
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/lib/vite-plugin-storybook-nextjs/src/plugins/next-mocks/alias/safe-stable-stringify/index.ts:15]: Won't fix for this PR.

The lossy `JSON.stringify` shim is an intentional CJS/`__dirname` workaround. Full `safe-stable-stringify` contract (circular / BigInt / key sort) is a heavy lift and out of scope here.

File: code/lib/vite-plugin-storybook-nextjs/src/plugins/next-swc/plugin.test.ts [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+
+vi.mock('../../utils/nextjs', () => ({
+  findNextDirectories: vi.fn(),
+  loadClosestPackageJson: vi.fn(),
+  loadSWCBindingsEagerly: vi.fn(),
+}));
+
+vi.mock('../../utils/swc/transform', () => ({
+  getVitestSWCTransformConfig: vi.fn(),
+}));
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/lib/vite-plugin-storybook-nextjs/src/plugins/next-swc/plugin.test.ts:22]: Won't fix for this PR.

Same as the other spy-mocking nits: style-only across several factory mocks, no failing behavior. Defer to a package-wide Vitest cleanup.

File: code/lib/vite-plugin-storybook-nextjs/src/plugins/next-swc/plugin.test.ts [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+      code: 'export {}',
+      map: null,
+    });
+
+    const swcTransform = await import('../../utils/swc/transform.ts');
+    vi.mocked(swcTransform.getVitestSWCTransformConfig).mockResolvedValue(
+      // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+      {} as any
+    );
+  };
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/lib/vite-plugin-storybook-nextjs/src/plugins/next-swc/plugin.test.ts:76]: Won't fix for this PR.

Per-test `setupMocks()` after `vi.resetModules()` is valid here. Moving setup into `beforeEach` is preference, not a bug.

File: code/lib/vite-plugin-storybook-nextjs/src/utils/nextjs.ts [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+
+  const mainPath = attempts === 1 ? ['.'] : new Array(attempts).fill('..');
+
+  try {
+    const file = await fs.promises.readFile(join(dir, ...mainPath, 'package.json'), 'utf8');
+    return JSON.parse(file);
+  } catch (e) {
+    return loadClosestPackageJson(dir, attempts + 1);
+  }
+}
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/lib/vite-plugin-storybook-nextjs/src/utils/nextjs.ts:56]: Fixed in 14f9257ecdb.

`attempts=2` now reads `../package.json` (was `../../`). Added a unit test for the parent-directory hit.

File: code/core/vitest.config.ts [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+    // oxc-parser / oxc-resolver ship `browser` entry points that import WASM bindings which
+    // are not installed in this monorepo. Force Vite/Vitest to resolve them via their Node
+    // entry points (index.js) during tests and benchmarks.
+    resolve: {
+      conditions: ['node'],
+      mainFields: ['main'],
+      // Nested @polka/url@0.5.0 is CJS and breaks polka's named `parse` import.
+      // Absolute path: `@polka/url/build.mjs` is not in the package exports map.
+      alias: {
+        '@polka/url': polkaUrlEsm,
```
Comment (JReinhold) [inline code/core/vitest.config.ts:23]: looks to me like there is a "conflict" between the polka version core uses, and polka used in this package? I think we should explore upgrading to the 1.0.0 prereleases that core also uses.

**Review by JReinhold — COMMENTED** [review]
Looks like _a lot_ of tests and stories were dropped from the original library during the move, when looking at the diff: https://github.com/storybookjs/storybook/pull/35833/changes/4b6f3f567ca410055c7b8ae93b48f430b858ce93..14f9257ecdbf608f0d1a7a15ddcd1c9dc8aaad92

was that intentional, because those cases are already covered within the `@storybook/nextjs-vite` framework package?

**Comment by ndelangen** [comment]
@JReinhold, yes, dropping the example stories/tests was intentional.

**Kept:** package unit tests under `src/**` (`index`, next-image, next-swc, next-config, nextjs utils). Those still run in `tests--linux`.

**Dropped:** the standalone `example/` Storybook app (and its stories/tests), plus standalone-repo tooling. 

In the monorepo, it should still be tested through frameworks: `@storybook/nextjs-vite` template stories via sandboxes / Chromatic / e2e.

Rough mapping vs `code/frameworks/nextjs-vite/template/stories/`:

| Dropped example story | In nextjs-vite templates? |
| --- | --- |
| Image / ImageLegacy / GetImageProps | yes |
| Font | yes |
| Link | yes |
| DynamicImport | yes |
| NavigationExports | mostly via `Navigation` / `Redirect` / `Router` |
| Header (`next/headers`) | `NextHeader` / related |
| Button / Header / Page (CSF starters) | CLI templates only — not feature coverage |
| Cache | **no direct equivalent** |
| EnvironmentVariables | **no direct equivalent** |
| UnstableRethrow | **no direct equivalent** |
| Tailwind / CSSModules / Sass / BootstrapButton | **no** (styling playground) |

Do you believe the skipped ones are real gaps in our testing (`Cache`, `EnvironmentVariables`, `UnstableRethrow`, …)?

I can likely write them into `code/frameworks/nextjs-vite/template/stories/`, if you wish

File: code/core/vitest.config.ts (hunk 16 above)
Comment (ndelangen) [reply code/core/vitest.config.ts:23]: Good catch on the churn here — but this isn’t a conflict with `vite-plugin-storybook-nextjs` (that package doesn’t depend on `polka` at all).

Core already uses `polka@^1.0.0-next.28`. The Vitest alias / inline + the root resolution `polka@npm:1.0.0-next.28/@polka/url` were added because Linux CI was nesting **`@polka/url@0.5.0`** (from **`polka@0.5.2` via `@compodoc/compodoc`**) under `node_modules/polka/`, which breaks polka@1’s ESM `import { parse } from '@polka/url'` and showed up as `tests--linux` + sandbox `--dev` failures on this PR (yarn install graph shifted when the new workspace package landed).

So the “upgrade to 1.0 prereleases” path would really mean getting rid of the **compodoc → polka@0.5.2** nest (upgrade/replace compodoc, or otherwise stop pulling 0.5.x), not changing anything in this plugin. Happy to do that as a separate cleanup if you want; the aliases are the minimal hardening so core doesn’t break when that nest appears.

**Comment by ndelangen** [comment]
Can we remove CJS support specific code?
As SB10 no longer supports CJS.

We should also add the stories:

Cache | no direct equivalent
EnvironmentVariables | no direct equivalent
UnstableRethrow | no direct equivalent

**Comment by ndelangen** [comment]
Follow-up on removing CJS support:
There are really two different things mixed under "CJS support":

**1. Shipping / consuming this package as CJS**
That was the hard dual-publish work: tsup building both ESM and CJS, `import`/`require` export conditions, `browser/mocks` as `.js` vs `node/mocks` as `.cjs`, and nextjs-vite having to `require()` the plugin because the ESM build was broken.

**That is already gone in this PR.** We only publish ESM now, nextjs-vite imports it normally, and there are no `.cjs` artifacts left in `dist/`.

**2. Interop with Next.js itself (still largely CJS)**
Things like unwrapping `default` from `next/dist/*` and `@next/env`, SWC emitting CJS for `next/dist` / `*.cjs`, `createRequire` to resolve Next's compiled paths, next/image CJS default quirks, and explicit re-exports for `next/navigation` / `next/router` because `export *` from CJS does not work for static ESM.

**That has to stay.** Removing it would break Next in Storybook, not "drop unused CJS support."

**Leftover that is actually simplify-able (follow-up, not this PR):**
`./browser/mocks/*` and `./node/mocks/*` now point at the same `.js` files. Collapsing that dual export surface is the remaining cleanup. Worth doing, but it needs its own Vitest browser/node pass, so I would not mix it into this move PR.

**So for this PR:** no further CJS purge. We already removed the package-level dual-publish work. Next interop stays. Mock export collapse can be a follow-up.

File: code/frameworks/nextjs-vite/template/stories/EnvironmentVariables.stories.tsx [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+export const Default: Story = {
+  play: async ({ canvas }) => {
+    await expect(canvas.getByTestId('nextConfigEnv')).toHaveTextContent('next-config-env');
+    await expect(canvas.getByTestId('nextPrefixEnv')).toHaveTextContent('example1');
+  },
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/frameworks/nextjs-vite/template/stories/EnvironmentVariables.stories.tsx:20]: Done in d44a69c59ab.

The original example left this assertion commented out. The story exists to prove `EXAMPLE2` stays out of the client bundle, so `nonNextPrefixEnv` now asserts `RESTRICTED_VALUE`.

File: code/frameworks/nextjs-vite/template/stories/UnstableRethrow.stories.tsx [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+
+type Story = StoryObj<typeof meta>;
+
+export const Default: Story = {};
+
+export const WithCatch: Story = {
+  args: {
+    shouldCatch: true,
+  },
+};
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/frameworks/nextjs-vite/template/stories/UnstableRethrow.stories.tsx:25]: Won't add play assertions here.

`WithCatch` is meant to exercise `unstable_rethrow` after a thrown error, not a heading change. Both stories render the same visible copy; asserting the heading would not distinguish the two cases. Default is a smoke render; `WithCatch` is the catch/rethrow path.

File: code/lib/vite-plugin-storybook-nextjs/README.md [hunk 19]
```diff
@@ -0,0 +1,197 @@
...
+export default {
+  framework: {
+    name: '@storybook/nextjs-vite',
+    options: {
+      image: {
+        // Default image glob patterns are already applied; this is optional
+        // include: ['**/*.{png,jpg,jpeg,gif,webp,avif,ico,bmp,svg}'],
+        // Ensure SVGR takes precedence for React components from SVGs
+        exclude: ['**/*.svg?icon'],
+      },
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/lib/vite-plugin-storybook-nextjs/README.md:103]: Done in d44a69c59ab.

The Storybook example now excludes `**/*.svg?react`, matching the FAQ text and the non-Storybook snippet.

File: code/lib/vite-plugin-storybook-nextjs/src/plugins/next-font/utils/set-font-declarations-in-head.ts [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+
+type Props = {
+  id: string;
+  fontFaceCSS: string;
+  classNamesCSS: string;
+};
+
+export function setFontDeclarationsInHead({ id, fontFaceCSS, classNamesCSS }: Props) {
+  // fontFaceCSS has placeholders for font path and fontReferenceId
+  // I want to extract them
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/lib/vite-plugin-storybook-nextjs/src/plugins/next-font/utils/set-font-declarations-in-head.ts]: Done in d44a69c59ab.

Deleted the investigation comments. The placeholder replacement below is enough.

File: code/lib/vite-plugin-storybook-nextjs/src/plugins/next-image/plugin.ts [hunk 21]
```diff
@@ -0,0 +1,231 @@
...
+      const [source, queryA] = id.split('?');
+
+      if (queryA === 'ignore') {
+        return null;
+      }
+
+      // For SVG files, only process if they don't have ?react parameter and SVG processing is enabled
+      const isSvg = /\.svg$/.test(source);
+      if (isSvg && hasVitePluginSvgr && queryA === 'react') {
+        return null;
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/lib/vite-plugin-storybook-nextjs/src/plugins/next-image/plugin.ts:103]: Won't fix in this PR.

`includePattern` already requires a trailing image extension. `#fragment` SVG sprite imports are a real edge case, but not something this move introduced, and changing `resolveId` matching needs its own image-plugin tests. Leaving as a follow-up.

File: scripts/tasks/sandbox-parts.ts [hunk 22]
```diff
@@ -347,6 +347,10 @@ export const init: Task['run'] = async (
     case '@storybook/angular-vite':
       await prepareAngularSandbox(cwd, template.name);
       break;
+    case '@storybook/nextjs':
+    case '@storybook/nextjs-vite':
+      await prepareNextjsSandbox(cwd);
+      break;
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply scripts/tasks/sandbox-parts.ts:355]: Done in d44a69c59ab.

`prepareNextjsSandbox` now runs only when `dryRun` is false.

File: scripts/tasks/sandbox-parts.ts [hunk 23]
```diff
@@ -1049,6 +1053,40 @@ async function prepareReactNativeWebSandbox(cwd: string) {
   }
 }
 
+// Env fixtures for nextjs template EnvironmentVariables stories.
+async function prepareNextjsSandbox(cwd: string) {
+  await writeFile(
+    join(cwd, '.env'),
+    ['NEXT_PUBLIC_EXAMPLE1=example1', 'EXAMPLE2=example2', ''].join('\n')
+  );
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply scripts/tasks/sandbox-parts.ts]: Done in d44a69c59ab.

The helper now reads an existing `.env` and upserts only `NEXT_PUBLIC_EXAMPLE1` and `EXAMPLE2`.

File: scripts/tasks/sandbox-parts.ts [hunk 24]
```diff
@@ -1049,6 +1053,40 @@ async function prepareReactNativeWebSandbox(cwd: string) {
...
+  if (!/nextConfig\s*(?::\s*NextConfig\s*)?=\s*\{/.test(source)) {
+    return;
+  }
+
+  await writeFile(
+    configPath,
+    source.replace(
+      /(nextConfig\s*(?::\s*NextConfig\s*)?=\s*\{)/,
+      `$1\n  env: { nextConfigEnv: 'next-config-env' },`
+    )
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply scripts/tasks/sandbox-parts.ts:1112]: Partially done in d44a69c59ab.

create-next-app configs typically have no `env` block, and parsing/merging a real `env` object from TS/JS is more machinery than this fixture needs. If `env` is already present, we now skip the insert instead of adding a duplicate property that would lose `nextConfigEnv`.

**Review by JReinhold — APPROVED** [review]
